cmd/vulcan-trivy/vulcan-trivy.go

```diff
@@ -103,65 +176,200 @@ func run(ctx context.Context, target, assetType, optJSON string, state checkstat
+		}
+
+		isReachable, err := helpers.IsReachable(target, assetType,
+			helpers.NewDockerCreds(os.Getenv("TRIVY_USERNAME"), os.Getenv("TRIVY_PASSWORD")))
+		if err != nil {
+			logger.Warnf("Can not check asset reachability: %v", err)
+		}
+		if !isReachable {
+			return checkstate.ErrAssetUnreachable
+		}
+
+		results, err := execTrivy(opt, "image", append(trivyArgs, target))
+		if err != nil {
+			return err
+		}
+
+		vuln := report.Vulnerability{
+			// Issue attributes.
+			Summary:     "Outdated Packages in Docker Image",
+			Description: "Vulnerabilities have been found in outdated packages installed in the Docker image.",
+			Recommendations: []string{
+				"Update affected packages to the versions specified in the resources table or newer.",
+			},
+			Details: strings.Join([]string{
+				"Run the following command to obtain the full report in your computer.",
+				"If using a public docker registry:",
+				fmt.Sprintf(`docker run -it --rm aquasec/trivy image %s`, target),
+				"\n",
+				"If using a private docker registry:",
+				fmt.Sprintf(`docker run -it --rm \
+				-e TRIVY_AUTH_URL=https://%s \
+				-e TRIVY_USERNAME=$REGISTRY_USERNAME \
+				-e TRIVY_PASSWORD=$REGISTRY_PASSWORD \
+				aquasec/trivy image %s`, registryEnvDomain, target),
+			}, "\n"),
+			CWEID:  937,
+			Labels: []string{"potential", "docker"},
+			// Finding attributes.
+		}
+		if err = processVulns(results.Results, vuln, "", state); err != nil {
+			logger.Errorf("processing image vuln results: %+v", err)
+		}
+
+		vuln = report.Vulnerability{
+			Summary:       "Secret Leaked in DockerImage",
+			Description:   "A secret has been found stored in the DockerImage. This secret could be retrieved by anyone with access to the image. Test data and false positives can be marked as such.",
+			CWEID:         540,
+			Score:         8.9,
+			ImpactDetails: "Anyone with access to the image could retrieve the leaked secret and use it in the future with malicious intent.",
+			Labels:        []string{"issue"},
+			Recommendations: []string{
+				"Completely remove the secrets from the repository as explained in the references.",
+				"Encrypt the secrets using a tool like AWS Secrets Manager or Vault.",
+			},
+			References: []string{
+				"https://help.github.com/en/articles/removing-sensitive-data-from-a-repository",
+			},
+		}
+		if err := processSecrets(results.Results, vuln, target, "", state); err != nil {
+			logger.Errorf("processing image secret results: %+v", err)
+		}
+
+		return processMisconfigs(results.Results, target, "", state)
+
+	} else if assetType == "GitRepository" {
+
+		sc := checksToParam(opt.GitChecks)
+		if sc == "" {
+			logger.Warnf("No checks enabled for DockerImage")
+			return nil
+		}
+		trivyArgs = append(trivyArgs, []string{"--security-checks", sc}...)
+
+		if opt.Depth == 0 {
+			opt.Depth = DefaultDepth
+		}
+		repoPath, branchName, err := helpers.CloneGitRepository(target, opt.Branch, opt.Depth)
+		if err != nil {
+			return err
```

Comment:
I think we should follow same convention as DockerImage or other assettypes, [verify if the asset is reachable](https://github.com/adevinta/vulcan-check-sdk/blob/master/helpers/target.go#L289) and if is not, return inconclusive.
I realised that vulcan-semgrep is lakcing this but I think it would be convinient.

Reply:
good point